src/benchmarks/micro/libraries/System.IO.FileSystem/Perf.FileStream.cs

```diff
@@ -2,152 +2,299 @@
+        [Arguments(OneMegaByte, FourKiloBytes, FileOptions.None)]
+        [Arguments(HundredMegaBytes, HalfKiloByte, FileOptions.None)]
+        [Arguments(HundredMegaBytes, FourKiloBytes, FileOptions.None)]
+        public long Read(long fileSize, int userBufferSize, FileOptions options)
         {
-            byte[] buffer = _buffer;
-            int bytesRead = 0;
-            
-            using (FileStream reader = new FileStream(_filePath, FileMode.Open, FileAccess.Read, FileShare.Read, DefaultBufferSize, FileOptions.Asynchronous))
+            byte[] userBuffer = _userBuffers[userBufferSize];
+            long bytesRead = 0;
+            using (FileStream fileStream = new FileStream(_sourceFilePaths[fileSize], FileMode.Open, FileAccess.Read, FileShare.Read, FourKiloBytes, options))
             {
-                for (int i = 0; i < TotalSize / BufferSize; i++)
+                while (bytesRead < fileSize)
                 {
-                    bytesRead += await reader.ReadAsync(buffer, 0, buffer.Length);
+                    bytesRead += fileStream.Read(userBuffer, 0, userBuffer.Length);
                 }
             }
 
             return bytesRead;
         }
 
         [Benchmark]
-        [BenchmarkCategory(Categories.NoWASM)]
-        public async Task CopyToAsync()
+        [Arguments(OneKiloByte, HalfKiloByte, FileOptions.None)]
+        [Arguments(OneKiloByte, FourKiloBytes, FileOptions.None)]
+        [Arguments(OneMegaByte, HalfKiloByte, FileOptions.None)]
+        [Arguments(OneMegaByte, FourKiloBytes, FileOptions.None)]
+        [Arguments(HundredMegaBytes, HalfKiloByte, FileOptions.None)]
+        [Arguments(HundredMegaBytes, FourKiloBytes, FileOptions.None)]
+        public void Write(long fileSize, int userBufferSize, FileOptions options)
         {
-            using (var reader = new FileStream(_filePath, FileMode.Open, FileAccess.Read, FileShare.Read, DefaultBufferSize, FileOptions.Asynchronous))
+            byte[] userBuffer = _userBuffers[userBufferSize];
+            using (FileStream fileStream = new FileStream(_destinationFilePaths[fileSize], FileMode.Create, FileAccess.Write, FileShare.Read, FourKiloBytes, options))
             {
-                await reader.CopyToAsync(Stream.Null);
+                for (int i = 0; i < fileSize / userBufferSize; i++)
+                {
+                    fileStream.Write(userBuffer, 0, userBuffer.Length);
+                }
             }
         }
 
         [Benchmark]
-        public void WriteByte()
+        [Arguments(OneKiloByte, HalfKiloByte, FileOptions.Asynchronous)]
+        [Arguments(OneKiloByte, FourKiloBytes, FileOptions.Asynchronous)]
+        [Arguments(OneMegaByte, HalfKiloByte, FileOptions.Asynchronous)]
+        [Arguments(OneMegaByte, FourKiloBytes, FileOptions.Asynchronous)]
+        [Arguments(HundredMegaBytes, HalfKiloByte, FileOptions.Asynchronous)]
+        [Arguments(HundredMegaBytes, FourKiloBytes, FileOptions.Asynchronous)]
```

Comment:
Only FileOptions.Asynchronous?

Reply:
@stephentoub Did you mean to use other flags like `FileOptions.RandomAccess`, `.FileOptions.SequentialScan` or `FileOptions.None`?

Reply:
None